### PR TITLE
Add io.github.jotd666.gods-deluxe

### PR DESCRIPTION
**Pull request by mbugni**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [X] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission

### CODE REVIEW

**Comment by bbhtt** [comment]
bot, build io.github.jotd666.gods-deluxe

File: io.github.jotd666.gods-deluxe.yaml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+modules:
+
+  # Install Java
+  - name: install-openjdk
+    buildsystem : simple
+    build-commands:
+      - /usr/lib/sdk/openjdk17/install.sh
+
+  # Install gods-deluxe from package
+  - name: gods-deluxe-build
```
Comment (bbhtt) [inline io.github.jotd666.gods-deluxe.yaml:28]: Is this x86-64 only? You have a bunch of x86_64 shared libraries in `/app/jre`. If aarch64 is not supported add a flathub.json

**Comment by bbhtt** [comment]
Seems like a missing library:

```
Using assets path: /app/share/gods-deluxe/
Using data path: /home/wirt/.var/app/io.github.jotd666.gods-deluxe/data/gods-deluxe/
java.io.FileNotFoundException: /home/wirt/.var/app/io.github.jotd666.gods-deluxe/data/gods-deluxe/gods_settings (No such file or directory)
	at java.base/java.io.FileInputStream.open0(Native Method)
	at java.base/java.io.FileInputStream.open(FileInputStream.java:216)
	at java.base/java.io.FileInputStream.<init>(FileInputStream.java:157)
	at java.base/java.io.FileInputStream.<init>(FileInputStream.java:111)
	at java.base/java.io.FileReader.<init>(FileReader.java:60)
	at gods.sys.ParameterParser.<init>(ParameterParser.java:280)
	at gods.sys.ParameterParser.open(ParameterParser.java:314)
	at gods.base.GameOptions.load_settings(GameOptions.java:226)
	at gods.game.GodsGame.main(GodsGame.java:321)
error loading shared library joystick.dll/.so: java.lang.UnsatisfiedLinkError: no joystick in java.library.path: /app/lib/gods-deluxe
```

**Comment by mbugni** [comment]
> Seems like a missing library:
> 

That library is available for Windows only.

File: io.github.jotd666.gods-deluxe.yaml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        strip-components: 0
+      - type: script
+        commands:
+          - 'exec /app/jre/bin/java -Djava.library.path="/app/lib/gods-deluxe" -DGODS_ASSETS_DIR="/app/share/gods-deluxe" -DGODS_DATA_DIR="${XDG_DATA_HOME}/gods-deluxe" -jar "/app/lib/gods-deluxe/gods-deluxe.jar" "$@"'
+        dest-filename: gods-deluxe.sh
+    post-install:
+      - install -Dm 755 gods-deluxe.sh /app/bin/gods-deluxe.sh
+
+    build-commands:
+      # Create target folders
```
Comment (hfiguiere) [inline io.github.jotd666.gods-deluxe.yaml:43]:
```suggestion
    build-commands:
      - install -Dm 755 gods-deluxe.sh /app/bin/gods-deluxe.sh
      # Create target folders
```

File: io.github.jotd666.gods-deluxe.yaml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+      # Copy assets
+      - cp --recursive ./fonts  --target-directory /app/share/gods-deluxe
+      - cp --recursive ./icons  --target-directory /app/share/gods-deluxe
+      - cp --recursive ./images --target-directory /app/share/gods-deluxe
+      - cp --recursive ./levels --target-directory /app/share/gods-deluxe
+      - cp --recursive ./locale --target-directory /app/share/gods-deluxe
+      - cp --recursive ./music  --target-directory /app/share/gods-deluxe
+      - cp --recursive ./sound  --target-directory /app/share/gods-deluxe
+      - cp --recursive ./tiles  --target-directory /app/share/gods-deluxe
+      - cp LICENSE --target-directory /app/share/gods-deluxe
```
Comment (hfiguiere) [inline io.github.jotd666.gods-deluxe.yaml:57]:
```suggestion
      - install -Dm644 LICENSE -t /app/share/gods-deluxe
      # Copy executables
      - cp --recursive ./bin/* --target-directory /app/lib/gods-deluxe
      # Copy assets
      - cp --recursive ./fonts  --target-directory /app/share/gods-deluxe
      - cp --recursive ./icons  --target-directory /app/share/gods-deluxe
      - cp --recursive ./images --target-directory /app/share/gods-deluxe
      - cp --recursive ./levels --target-directory /app/share/gods-deluxe
      - cp --recursive ./locale --target-directory /app/share/gods-deluxe
      - cp --recursive ./music  --target-directory /app/share/gods-deluxe
      - cp --recursive ./sound  --target-directory /app/share/gods-deluxe
      - cp --recursive ./tiles  --target-directory /app/share/gods-deluxe
```

**Comment by hfiguiere** [comment]
bot, build io.github.jotd666.gods-deluxe

**Pull request closed without merging**